### PR TITLE
docs: Improve code-block highlighting

### LINKED ISSUE #22050: Improve code block higlighting colors in docs
Currently the color scheme for code block highlighting just doesnt look right to me:

code-block-colors

while i dont think red and green should _never_ be seen its probably not the best visual contrast and is known to be troubling to people with color blindness and/or dyslexia

i also think red should generally not be used for anything other than errors

### PR DESCRIPTION
**Pull request by phlax**
Fix #22050 

This PR links docs builds with a selection of pygments themes for code-block highlighting

We can adapt a chosen theme, but the less we have to do so the better

Some changes are easier than others - for example changing the background

I think that a few of the themes might work better if they didnt have a white background

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

#### pygments previews

https://dt.iki.fi/pygments-gallery

#### good pages to view

- start/quick-start/run-envoy.html
- start/sandboxes/ext_authz.html
- start/sandboxes/wasm-cc.html
- api-v3/config/bootstrap/v3/bootstrap.proto.html#config-bootstrap-v3-bootstrap
- api-v3/config/route/v3/route_components.proto.html
- docs/operations/admin.html
- intro/arch_overview/advanced/matching/matching_listener.html

#### themes currently available in the build

```python
['default', 'emacs', 'friendly', 'friendly_grayscale', 'colorful', 'autumn', 'murphy', 'manni', 
 'material', 'monokai', 'perldoc', 'pastie', 'borland', 'trac', 'native', 'fruity', 'bw', 'vim', 'vs',
 'tango', 'rrt', 'xcode', 'igor', 'paraiso-light', 'paraiso-dark', 'lovelace', 'algol', 'algol_nu',
 'arduino', 'rainbow_dash', 'abap', 'solarized-dark', 'solarized-light', 'sas', 'stata',
 'stata-light', 'stata-dark', 'inkpot', 'zenburn', 'gruvbox-dark', 'gruvbox-light', 'dracula',
 'one-dark', 'lilypond']
```

### CODE REVIEW

**Comment by phlax** [comment]
#### solarized-light

https://storage.googleapis.com/envoy-pr/be9d036/docs/index.html


- works well in API i think

**Comment by phlax** [comment]
##### gruvbox-light

https://storage.googleapis.com/envoy-pr/c78c00c/docs/index.html

- better contrast/definition than solarized-light
- emphasis-highlighting within code-blocks is fu

**Comment by phlax** [comment]
#### borland

https://storage.googleapis.com/envoy-pr/b2b4bc4/docs/index.html

**Comment by phlax** [comment]
#### tango

https://storage.googleapis.com/envoy-pr/918ad62/docs/index.html

- in some ways i like this one the most so far, but there are some bits that look odd

**Comment by phlax** [comment]
#### lovelace

https://storage.googleapis.com/envoy-pr/82f1844/docs/index.html

**Comment by phlax** [comment]
#### rainbow_dash

https://storage.googleapis.com/envoy-pr/617d52e/docs/index.html

- not too keen on this one - thought the green/purple might work for Envoy but doesnt  look great

**Comment by phlax** [comment]
#### arduino

https://storage.googleapis.com/envoy-pr/afca68c/docs/index.html

- quite plain
- could be adapted ~easily

**Comment by phlax** [comment]
#### stata

https://storage.googleapis.com/envoy-pr/516c788/docs/index.html

**Comment by phlax** [comment]
#### manni

https://storage.googleapis.com/envoy-pr/b269afa/docs/index.html

**Comment by phlax** [comment]
#### friendly

https://storage.googleapis.com/envoy-pr/4a36ac1/docs/index.html

- this one i really like!

**Comment by moderation** [comment]
+1 for tango

**Comment by phlax** [comment]
looking over tango again - i think my main gripe is the console logging - i reckon i can fix and perhaps adjust a few of the colors

lets use that one as the base

File: docs/conf.py [hunk 1]
```diff
@@ -22,6 +22,10 @@
 import sphinx_rtd_theme
 
 
+# TODO(phlax): move the pytments style to envoy.docs.sphinx_runner and remove this
```
Comment (phlax) [inline docs/conf.py:25]: pygments

**Comment by phlax** [comment]
@moderation @mattklein123 PTAL - i have used the tango theme, with 2 changes

- made the highlight background less bright
- de-italicised and softened console output


its not perfect, but i think better than what we have and probs good enough for now

**Comment by mattklein123** [comment]
> de-italicised and softened console output

If I'm really nit picking the console output is a bit light for my eyes? Can we darken at a bit? I agree on not italicizing it. Everything else looks awesome from a quick skim.

With that said definitely happy to be outvoted on the darkness.

**Comment by phlax** [comment]
i moved it up a couple of stops in the grayscale `#aaa` -> `#888`

**Comment by phlax** [comment]
any more and it loses distinction from console text

**Review by mattklein123 — APPROVED** [review]
LGTM! Nice!

**Pull request merged**